src/test/com/yetanalytics/datasim/xapi/profile_test.clj

```diff
@@ -289,26 +290,96 @@
   (s/cat :satisfieds cmi5-satisfieds?
          :typical-sessions cmi5-typical-sessions?))
 
+(defn walk-pattern* [profiles alignments seed]
+  (let [{:keys [pattern-walk-fn]}
+        (profile/profiles->profile-map profiles {} 100)]
+    (pattern-walk-fn alignments (random/seed-rng seed))))
+
 (s/fdef walk-pattern
   :args (s/cat :seed int?)
   :ret cmi5-general-pattern?)
 
 (defn walk-pattern [seed]
-  (let [{:keys [profiles alignments]} const/simple-input
-        alignment   (->> (get-in alignments [:alignment-vector 0 :alignments])
-                         (reduce (fn [m {:keys [component] :as align}]
-                                   (assoc m component align))
-                                 {}))
-        {:keys [pattern-walk-fn]}
-        (profile/profiles->profile-map profiles {} 100)]
-    (pattern-walk-fn alignment (random/seed-rng seed))))
+  (let [{:keys [profiles models]} const/simple-input
+        alignments (->> (get-in models [0 :alignments])
+                        (reduce (fn [m {:keys [id weights]}]
+                                  (assoc m id weights))
+                                {})
+                        (assoc {} :weights))]
+    (walk-pattern* profiles alignments seed)))
 
 (deftest walk-pattern-test
   (testing "Walk and generate seq for a single pattern"
     (let [{total :total check-passed :check-passed}
           (stest/summarize-results (stest/check `walk-pattern))]
       (is (= total check-passed)))))
 
+(deftest walk-weighted-pattern-test
+  (testing "Remove abandoned template from consideration"
+    (let [{:keys [profiles]} const/simple-input
+          weights (-> {"terminated" 1.0
+                       "abandoned"  0.0}
+                      (update-keys cmi5-iri))
+          results (walk-pattern* profiles {:weights weights} 100)]
+      (is (s/valid? cmi5-general-pattern? results))
+      (is (s/valid? cmi5-terminated? (last results)))
+      (is (not (s/valid? cmi5-abandoned? (last results))))))
+  (testing "Remove terminated template from consideration"
+    (let [{:keys [profiles]} const/simple-input
+          weights (-> {"terminated" 0.0
+                       "abandoned"  1.0}
+                      (update-keys cmi5-iri))
+          results (walk-pattern* profiles {:weights weights} 100)]
+      (is (s/valid? cmi5-general-pattern? results))
+      (is (s/valid? cmi5-abandoned? (last results)))
+      (is (not (s/valid? cmi5-terminated? (last results))))))
+  (testing "Force completed pattern to appear"
+    ;; FIXME: This does not guarentee that "completed" appears, since
+    ;; the weight for `nil` is still 0.5, not 0.0
```

Comment:
I realized this is a problem, but a) this issue was present with the old alignments weights and b) IDK if we actually need to fix this?